spread/client.go

```diff
@@ -1011,7 +1011,7 @@ func outputErr(output []byte, err error) error {
 	return err
 }
 
-func waitPortUp(ctx context.Context, what fmt.Stringer, address string) error {
+func waitPortUp(ctx context.Context, what fmt.Stringer, address string, cmd *exec.Cmd) error {
```

Comment:
Instead of adding a extra parameter here we need to teach the call site to interrupt ctx when cmd dies. So a go-routine to monitor cmd and cancel the context.